src/hardware/sblaster.cpp

```diff
@@ -125,6 +121,22 @@ enum class DmaMode {
 
 enum class EssType { None, Es1688 };
 
+enum class FrameType { Mono, Stereo };
+
+class Dac {
+public:
+	int MeasureHighwaterRateHz();
+	AudioFrame RenderFrame();
+
+private:
+	double last_write_ms   = {};
+	int rate_change_tally  = {};
+	int high_water_rate_hz = MinPlaybackRateHz;
+};
+
+// TODO: Combine the SBLASTER class, 'sb' global struct, and all the static
+// functions into a single unified class with member data and member functions.
```

Comment:
General comment: there is _a lot_ happening in this commit! In your commit message you're listing 5 individual changes. I know it's a hassle, but would it be possible to at least move the mechanical code-shuffling out of this huge commit for easier bisecting? If this commit breaks something, well, good luck hunting down the offending line  a few months later... 😆

Reply:
A lot of it was coupled; but will try to move out as much as possible into separate commits.